src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,53 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar-width;
+  gap: $gap-size;
+
+  $days-of-week:
+    "mon" 1,
+    "tue" 2,
+    "wed" 3,
+    "thu" 4,
+    "fri" 5,
+    "sat" 6,
+    "sun" 7;
+
+  @each $start-day, $start-day-value in $days-of-week {
+    &--start-day--#{$start-day} :first-child {
+      margin-left: ($start-day-value - 1) * ($cell-size + $gap-size);
+    }
+  }
+
+  @for $daysCount from 28 through 31 {
+    &--month-length--#{$daysCount} {
+      .calendar__day:nth-child(n + #{$daysCount + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: $cell-size;
+    height: $cell-size;
+    background-color: #eee;
+    border: 1px solid #000;
+    cursor: pointer;
+
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day})::before {
+        content: "#{$day}";
+        @include text-styles;
+      }
+    }
+
+    &:hover {
+      background-color: pink;
+      transform: translateY(-20px);
+      transition: 0.4s transform;
```

Comment:
Transition property should be under the general selector (9th item in the checklist). Like `&__day: {transition: color 0.5s, transform 0.5s;}`

Reply:
when i do this tast i didn'n saw lessons about those properties...